src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -16,6 +16,34 @@ public Rectangle(Integer width, Integer length, String color) {
         this.color = color;
     }
 
+    @Override
+    public int hashCode() {
+        if (width == null || length == null || color == null) {
+            return 0;
+        }
+        int res = 17;
+        res = 31 * res + width.hashCode();
+        res = 31 * res + length.hashCode();
+        res = 31 * res + color.hashCode();
+        return res;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (width == null || length == null || color == null || obj == null) {
+            return true;
+        } else if (obj == this) {
```

Comment:
we don't have to use else block after return statement cause it will exit from method